code/modules/mob/living/silicon/robot/robot_model.dm

```diff
@@ -45,7 +45,7 @@
 	var/allow_riding = TRUE
 	///Whether the borg can stuff itself into disposals
 	var/canDispose = FALSE
-	///The y offset of  the hat put on
+	///The y offset of the hat put on for symmetrical borgs or a list of "north" "south" "east" "west" x,y offsets for asymmetrical ones
 	var/hat_offset = -3
```

Comment:
I can make every TG borg use the offset list, but I'm not going to break the compatibility with using a number for it since there's no way I'm going to redo all the hat_offsets downstream to be lists.

Reply:
Tophat uses infinity for it as well to prevent putting a hat on it `"Tophat" = list(SKIN_ICON_STATE = "tophat", SKIN_LIGHT_KEY = NONE, SKIN_HAT_OFFSET = INFINITY),`

Reply:
This code suggestion doesn't cover the individual offsets of every borg skin.

Reply:
Fuck it... I'll just regex replaceall

Reply:
Regex is king